packages/unhead/src/composables/useScript.ts

```diff
@@ -143,8 +141,24 @@ export function useScript<T extends Record<symbol | string, any> = Record<symbol
     },
     updateTrigger(trigger: UseScriptOptions['trigger']) {
       // cancel previous trigger
-      triggerAbortController.abort()
-      handleTrigger(trigger, triggerAbortController, script, head)
+      script._triggerAbortController?.abort()
+      if (((typeof trigger === 'undefined' || trigger === 'client') && !head.ssr) || trigger === 'server') {
+        script.load()
+      }
+      else if (trigger instanceof Promise) {
+        script._triggerAbortController = new AbortController()
+        Promise.race([
```

Comment:
Love the Promise.race here ! :heart:

Reply:
Thanks!

Your implementation was good but I was having trouble figuring out what would happen if the abort was called while the script load promise was pending